engine/runtime/src/main/java/org/enso/interpreter/node/ProgramRootNode.java

```diff
@@ -62,6 +62,9 @@ public Object execute(VirtualFrame frame) {
         module = new Module(name, null, sourceCode.getCharacters().toString());
       }
       ctx.getPackageRepository().registerModuleCreatedInRuntime(module.asCompilerModule());
+      if (ctx.isStrictErrors()) {
+        module.compileScope(ctx);
```

Comment:
Truffle TCK would like us to throw the _syntax error_ from `ctx.eval` as a result of that we seem to have [these test failures](https://github.com/enso-org/enso/actions/runs/8965563388/job/24619295460?pr=9864#step:7:4822):
```
org.enso.interpreter.test.SignatureTest
org.enso.interpreter.test.EqualsTest
org.enso.interpreter.test.HashCodeTest
org.enso.interpreter.test.tck.EnsoTckInvalidSyntaxTest
org.enso.interpreter.runtime.ModuleTest
org.enso.interpreter.node.expression.builtin.meta.TypeOfNodeTest
org.enso.interpreter.test.DiagnosticFormatterTest
org.enso.interpreter.test.IntegerTest
org.enso.interpreter.test.AtomConstructorTest
org.enso.interpreter.test.FindExceptionMessageTest
org.enso.interpreter.test.BinaryOpIntegerTest
org.enso.compiler.ExecStrictCompilerTest
org.enso.interpreter.test.BinaryOpFloatTest
```
those tests will need to be adjusted to the fact that `CompilationAbortedException` is thrown sooner than it used to be.

Reply:
Addressed by 0effcd7, 7ccab76 and 747fd5c